common/djangoapps/student/helpers.py

```diff
@@ -122,8 +124,16 @@ def check_verify_status_by_course(user, course_enrollment_pairs, all_course_mode
         # verification status.
         if verified_mode is not None and enrollment.mode in CourseMode.VERIFIED_MODES:
             deadline = verified_mode.expiration_datetime
+
             relevant_verification = SoftwareSecurePhotoVerification.verification_for_datetime(deadline, verifications)
 
+            # Picking the max verification datetime on each iteration
+            if relevant_verification is not None:
+                recent_verification_datetime = max(
+                    recent_verification_datetime if recent_verification_datetime is not None
+                    else relevant_verification.expiration_datetime, relevant_verification.expiration_datetime
```

Comment:
Also, could you please add a test case for the scenario you described?  The issue is pretty subtle (I'm really glad you caught it!) and I want to make sure we don't accidentally this later.